src/figuresClasses.ts

```diff
@@ -1,19 +1,103 @@
+
+type Color = 'red' | 'green' | 'blue';
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number;
 }
 
-export class Triangle {
+function calculateFigureArea(value: number): number {
+  return Math.floor(value * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sidesSum = this.a + this.b + this.c;
+    const maxSide = Math.max(this.a, this.b, this.c);
+    const sumTwoSmallerSides = sidesSum - maxSide;
+
+    if (maxSide >= sumTwoSmallerSides) {
+      throw new Error('The largest side is less '
+        + 'than the sum of the two smaller sides');
+    }
 
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+    ) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
This message would be good to use in first condition, here better would be to inform user that side can't be < 0